src/url.js

```diff
@@ -0,0 +1,27 @@
+import isUrl from 'is-url';
+
+import wrapValidator from './helpers/wrapValidator';
+
+function requiredUrl(props, propName, componentName) {
```

Comment:
This validator should also check that `PropTypes.string` passes - `is-url` appears to coerce any input to a string, but we should be stricter than that.

Reply:
I wanted to use [`PropTypes.checkPropTypes`](https://github.com/facebook/prop-types#proptypescheckproptypes) for the `PropTypes.string` check but it throws on invalid prop types and adding a `try...catch` around the `PropTypes.checkPropTypes` method call did not seem very clean.

It looks like the `string` PropType is defined [here](https://github.com/facebook/react/blob/14156e56b9cf18ac86963185c5af4abddf3ff811/src/isomorphic/classic/types/ReactPropTypes.js#L79) where [`createPrimitiveTypeChecker`](https://github.com/facebook/react/blob/14156e56b9cf18ac86963185c5af4abddf3ff811/src/isomorphic/classic/types/ReactPropTypes.js#L191) does a `typeof` check on the prop value in [`getPropType`](https://github.com/facebook/react/blob/14156e56b9cf18ac86963185c5af4abddf3ff811/src/isomorphic/classic/types/ReactPropTypes.js#L480).

I replicated this behavior in [the `isString` helper](https://github.com/airbnb/prop-types/pull/35/files#diff-170ad113322ea13d9558272f6389be30R1). Let me know if creating a separate `typeof` helper is overkill and I can move the logic to `url.js`.

Reply:
It's supported if you pass `...rest` along with it.